### PR TITLE
Flask base improvements

### PR DESCRIPTION
**Pull request by sandlerben**
@anniemeng I think we should merge almost all of these improvements in, except perhaps the python auto-formatting. What do you think?

### CODE REVIEW

File: app/templates/layouts/base.html [hunk 1]
```diff
@@ -2,15 +2,39 @@
...
 <html>
-    {% include 'partials/_head.html' %}
-
+    <head>
+        {% include 'partials/_head.html' %}
+        {# Any templates that extend this template can set custom_head_tags to add scripts to their page #}
+        {% block custom_head_tags %}{% endblock %}
+    </head>
     <body>
+      {# Example dropdown menu setup. Uncomment lines to view
```
Comment (anniemeng) [inline app/templates/layouts/base.html]: maybe we should remove these examples for our specific project?

**Comment by anniemeng** [comment]
@sandlerben what do you think?

**Comment by sandlerben** [comment]
yes, agreed. I'm very busy but will take a look later this week

**Comment by sandlerben** [comment]
Adding all these changes isn't going to work, but I'll make another pr with the changes I think are important.

**Pull request closed without merging**